jlm/rvsdg/node.hpp

```diff
@@ -292,7 +292,7 @@ class Output
   [[nodiscard]] bool
   IsDead() const noexcept
   {
-    return NumUsers_ == 0;
+    return Users_.size() == 0;
```

Comment:
As the intrusive list keeps now track of the size of the list, it should be possible to remove the NumUsers_ attribute from the Output class.

Reply:
As per other comment, I would suggest not counting in the list ADT but outside